tt_metal/impl/buffers/buffer.cpp

```diff
@@ -16,6 +16,18 @@ namespace tt {
 
 namespace tt_metal {
 
+#if defined(TRACY_ENABLE)
+
+std::unordered_map<int, std::string> global_mempool_names;
+
+static const char * get_buffer_location_name (BufferType buffer_type, int device_id) {
+    std::string global_mempool_name = fmt::format("{} {}",magic_enum::enum_name(buffer_type), device_id);
+    int name_combo = (int)buffer_type * 1000 + device_id;
+    global_mempool_names.emplace(name_combo, global_mempool_name);
+    return global_mempool_names[name_combo].c_str();
```

Comment:
We should shuffle this around and use map::find to see if there is already an entry and return it, or else construct/emplace the string instead of always creating the string.

Reply:
@tt-asaigal does this need to be made thread-safe?

Reply:
Yes, in the current implementation of async each worker thread can call alloc/dealloc. This either needs to be made `thread_local` or be protected by a mutex